grass7/raster/r.vol.dem/chull.c

```diff
@@ -27,10 +27,6 @@
 
 #include "globals.h"
 
-/*Define Boolean type */
-typedef enum
-{ BFALSE, BTRUE } bool;
-
```

Comment:
With the following modification to existing PR, it should work with pre-7.8.6 GRASS too.

```diff
@@ -27,6 +27,12 @@
 
 #include "globals.h"
 
+#ifndef __bool_true_false_are_defined
+/* Define Boolean type for pre-7.8.6 compatibility */
+typedef enum
+{ false = FALSE, true = TRUE } bool;
+#endif
+
 /* Define vertex indices. */
 #define X   0
 #define Y   1
```

Small explanation: with 7.8.6 `stdbool.h` is included in `gis.h` (which is in turn included in `chull.c`). stdbool.h defines `__bool_true_false_are_defined = 1`: [stdbool.h specs](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/stdbool.h.html).

For 7.8.5 and earlier versions, stdbool.h is **not** included, `__bool_true_false_are_defined` is not true and local definition of bool is made.

*I have tested the above code and logic in test code, not with the actual extension.*

Reply:
Thanks a lot, @nilason - changes added in f82f1107